nautobot/core/tests/test_navigations.py

```diff
@@ -82,3 +87,84 @@ def test_permissions_rollup(self):
                 else:
                     expected_perms[tab_name] |= group_perms
             self.assertEqual(expected_perms[tab_name], tab_details["permissions"])
+
+    def test_nav_menu_tabs_have_icon_and_weight(self):
+        """Ensure each NavMenuTab in every navigation.py has an icon and weight set, and any duplicates by name match."""
+        tabs_by_name = {}
+        apps_dir = os.path.abspath(os.path.join(os.path.dirname(__file__), "..", ".."))
+        for folder in os.listdir(apps_dir):
+            folder_path = os.path.join(apps_dir, folder)
+            nav_path = os.path.join(folder_path, "navigation.py")
+            if not os.path.isdir(folder_path):
+                continue
+            try:
+                if not os.path.isfile(nav_path):
+                    continue
+                module_name = f"nautobot_{folder}_navigation"
+                spec = importlib.util.spec_from_file_location(module_name, nav_path)
+                nav_module = importlib.util.module_from_spec(spec)
+                spec.loader.exec_module(nav_module)
+            except (FileNotFoundError, ImportError, SyntaxError):
+                # Skip folders without a valid navigation.py or with import errors
+                continue
+            menu_items = getattr(nav_module, "menu_items", [])
+            if not menu_items:
+                continue
+            for tab in menu_items:
+                if not isinstance(tab, NavMenuTab):
+                    continue
```

Comment:
should this fail? Can't think of a use case for having something that isn't a tab here.